include/fx/gltf.h

```diff
@@ -798,6 +805,7 @@ namespace gltf
         int32_t scene{ -1 };
         std::vector<std::string> extensionsUsed{};
         std::vector<std::string> extensionsRequired{};
+        std::vector<detail::Chunk> extraChunks;

```

Comment:
The Document is a public facing object so it having a container of detail::Chunks seems wrong - public users shouldn't care about detail:: stuff.  Moreover the Chunks are pretty low level and I imagine many implementations will want to take the raw byte buffer and map it to their own structures(??).

Would a different design be better?  What if the file load/save paths had some extensibility points, like callbacks, that implementations could opt-in to?  For example, when loading, their callback would be called with information that points to the extra chunks and then they are free to process it directly instead of waiting for it to be packed into these Chunks here?